tests/Benchmark/EntryPoint.cpp

```diff
@@ -0,0 +1,146 @@
+            mean += (results[i] - mean) / (i + 1);
+            q += (results[i] - previousMean) * (results[i] - mean);
+        }
+        Microseconds sampleStandardDeviation = std::sqrt(q / (runCount - 1));
+        m_results.emplace_back(__Result{ testName, netRuntime, mean, sampleStandardDeviation });
+    }
+
+protected:
+    std::vector<__Result> m_results;
+};
+
+class CSVBenchmarkTestPublisher : public TestPublisherBase {
+    std::string m_outPath;
+
+public:
+    CSVBenchmarkTestPublisher(std::string&& outPath) : m_outPath(outPath) {
+    }
+
+    virtual void PublishTestResults() {
+        std::ofstream outFile(m_outPath);
+        outFile << "Test Name"
+                << ","
+                << "Total Runtime"
+                << ","
+                << "Mean Runtime"
+                << ","
+                << "Standard Deviation"
+                << "\n";
+        for (auto res : m_results) {
+            outFile << res.testName << "," << res.totalRuntime << "," << res.meanRuntime << "," << res.stdDeviation << "\n";
+        }
+
+        outFile.flush();
+        outFile.close();
+    }
+};
+
+class LogBenchmarkTestPublisher : public TestPublisherBase {
+    size_t m_maxNameWidth = 0;
+
+public:
+    virtual void RegisterTestResults(const std::string& testName, const std::vector<Microseconds>& results) {
+        m_maxNameWidth = max(m_maxNameWidth, testName.size());
+        TestPublisherBase::RegisterTestResults(testName, results);
+    }
+
+    virtual void PublishTestResults() {
+        // Align output horizontally
+        std::string spaces(m_maxNameWidth - 5, ' ');
+        std::cout << "Test Name" << spaces << "|"
+                  << "Total Runtime"
+                  << "\t\t|"
+                  << "Mean Runtime"
+                  << "\t\t|"
+                  << "Standard Deviation"
+                  << "\n";
+        for (auto res : m_results) {
+            std::string spaces(m_maxNameWidth - res.testName.size() + 4, ' ');
+            std::cout << std::fixed << res.testName << spaces << "|" << res.totalRuntime << "\t\t|" << res.meanRuntime << "\t\t|"
+                      << res.stdDeviation << "\n";
+        }
+
+        std::cout.flush();
+    }
+};
+
+static std::shared_ptr<BenchmarkTestPublisher> s_publisher;
+void BenchmarkTestCreator::CreatePublisher(int argc, char** argv) {
+    // Currently only support CSV format
+    for (int i = 1; i < argc && argv[i]; ++i) {
+        char* arg = argv[i];
+        if (strncmp(arg, "--out=", 6) == 0) {
+            s_publisher.reset(new CSVBenchmarkTestPublisher(std::move(std::string(arg + 6))));
+            return;
+        }
+    }
+
+    LOG_INFO("No arguments given to benchmark tests. Defaulting to logging results");
+    s_publisher.reset(new LogBenchmarkTestPublisher());
+}
+
+std::shared_ptr<BenchmarkTestPublisher> BenchmarkTestCreator::GetPublisher() {
+    return s_publisher;
+}
+
+int main(int argc, char** argv) {
+#ifdef WIN32
```

Comment:
We dont need this to run in reference platform, it that will make things easy w.r.t future integration to TAEF.